lcls-twincat-common-components/Library/POUs/FB_CC_TempSensor.TcPOU

```diff
@@ -0,0 +1,22 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1">
+  <POU Name="FB_CC_TempSensor" Id="{62e52992-2558-4a9c-b1dd-b3b9616cbbcf}" SpecialFunc="None">
+    <Declaration><![CDATA[FUNCTION_BLOCK FB_CC_TempSensor EXTENDS FB_TempSensor_FFO
+VAR
+    rtVetoReset: R_TRIG;
+END_VAR
+]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[// Database default read is zero if no entry
+bVeto S= fFaultThreshold = 0;
```

Comment:
What's the idea behind vetoing if this is 0?

Reply:
0 is the threshold we get if there is no entry in the database for reactive temperature.
Scientific consensus was that we should probably change the default number to something else but that's what it is for now
The idea was that we wouldn't force them to come up with temperatures for all the states, but maybe this is incorrect?

Reply:
Also, should I revise the comment maybe? I realize now my comment is unclear

Reply:
I was worried that there might be a use case where 0 is actually the threshold, but if its okay for now that works. The comment is okay.  I guess to me it's a bit counter-intuitive. If someone uses this FB they presumably want to fault given a threshold. If they aren't able to load one from the database shouldn't we fault?

Reply:
It sounded like, in the future, 0 could potentially be a threshold for a system that needs to really really be cooled. At present, it's not a realistic fault temperature so I ignored this for now.

So, we could change the default to either something really really low (to fault if nothing was entered) or to something really really high (to ignore if nothing was entered).

I think there are good philosophical reasons for either which way.

Reply:
After a few days of background-thread braining on this, I figure I should simplify the logic here to "trust what pmps gives me and treat it seriously". So, if PMPS says the temp threshold is 0, I treat it as zero.

Then, if I want different default behavior, I should solve this in lcls-twincat-pmps, where we can decide if "no temperature database entry" means "absolute zero" (always fault on any reading), or if "no entry" means "set the number really high" (so we never fault).